Client/Clique/Modules/EventDetail/EventDetailViewController.swift

```diff
@@ -20,32 +22,175 @@ class EventDetailViewController: UIViewController {
+            }
+        }
+    }
+    
+    func activateButton(){
+        eventStatus = didItEnd(timestamp: event.timestamp)
+        if(eventStatus == Event.status.pending){
+            cosmosView.isHidden = true
+            buttonJoinEvent.addTarget(self, action: #selector(registerToEventAlert), for: .touchUpInside)
+            buttonJoinEvent.isHidden = false
+            if(status == 1){
+                buttonJoinEvent.setTitle("Join", for: .normal)
+                alertMessage = Constants.Alerts.joinEventMessage
+            } else if(status == 2) {
+                buttonJoinEvent.setTitle("Leave", for: .normal)
+                alertMessage = Constants.Alerts.cancelEventMessage
+            } else if(status == 3) {
+                buttonJoinEvent.setTitle("You cannot rejoin event", for: .normal)
+                buttonJoinEvent.isEnabled = false
+            }
+        } else if(eventStatus == Event.status.done && status == 2) {
+            cosmosView.isUserInteractionEnabled = false
+            getRatedEvent()
+        } else {
+            buttonJoinEvent.isHidden = true
+        }
+    }
+    
+    func setCosmosRating() {
+        cosmosView.isHidden = false
+        buttonJoinEvent.isHidden = false
+        if(cosmosView.rating == 0) {
+            cosmosView.isUserInteractionEnabled = true
+            buttonJoinEvent.addTarget(self, action: #selector(rateEventAlert), for: .touchUpInside)
+            buttonJoinEvent.setTitle("Rate", for: .normal)
+        } else {
+            buttonJoinEvent.setTitle("Event already rated", for: .normal)
+            buttonJoinEvent.isEnabled = false
+            let rating = cosmosView.rating
+            cosmosView.isUserInteractionEnabled = false
+            cosmosView.rating = rating
+        }
+    }
+    
+    @objc func registerToEventAlert(){
+        let refreshAlert = UIAlertController(title: Constants.Alerts.confirmationTitleMessage, message: alertMessage, preferredStyle: UIAlertController.Style.alert)
+
+        refreshAlert.addAction(UIAlertAction(title: Constants.Alerts.defaultOKActionTitle, style: .default, handler: { (action: UIAlertAction!) in self.callRegisterToEventService()
+        }))
+
+        refreshAlert.addAction(UIAlertAction(title: Constants.Alerts.defaultCancelActionTitle, style: .cancel, handler: { (action: UIAlertAction!) in
+        }))
+
+        present(refreshAlert, animated: true, completion: nil)
+    }
+    
+    @objc func rateEventAlert(){
+        eventServices.rateEvent(event_id: event.id, rating: cosmosView.rating){[weak self] result in
+            guard let _self = self else { return }
+            switch result{
+            case .success(let success):
+                _self.sendOkAlert(message: success)
+                _self.setCosmosRating()
+            case .failure(_):
+                _self.sendOkAlert(message: Constants.Alerts.rateEventError)
+            }
+        }
+    }
+    
+    func callRegisterToEventService(){
```

Comment:
```suggestion
    func callRegisterToEventService() {
```

Formatting. Please fix this everywhere